benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -37,3 +41,46 @@ def time_make_grid(self):
         esmf_grid.data[:] = 0
 
     time_make_grid.version = 1
+
+
+class TimeRegridding:
+    params = ["similar", "large source", "large target", "mixed"]
+    param_names = ["source/target difference"]
+
+    def setup(self, type):
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        n_lons_src = 20
+        n_lats_src = 40
+        n_lons_tgt = 20
+        n_lats_tgt = 40
+        h = 100
+        if type == "large source":
+            n_lons_src = 100
+            n_lats_src = 200
+        if type == "large target":
+            n_lons_tgt = 100
+            n_lats_tgt = 200
+        if type == "mixed":
+            coord_system_src = RotatedGeogCS(0, 90, 90)
+        else:
+            coord_system_src = None
+        grid = _grid_cube(
+            n_lons_src,
+            n_lats_src,
+            lon_bounds,
+            lat_bounds,
+            coord_system=coord_system_src,
+        )
+        tgt = _grid_cube(n_lons_tgt, n_lats_tgt, lon_bounds, lat_bounds)
+        src_data = np.arange(n_lats_src * n_lons_src * h).reshape(
+            [n_lats_src, n_lons_src, h]
+        )
+        src = Cube(src_data)
+        src.add_dim_coord(grid.coord("latitude"), 0)
+        src.add_dim_coord(grid.coord("longitude"), 1)
+        self.regridder = ESMFAreaWeightedRegridder(src, tgt)
+        self.src = src
```

Comment:
Should we be timing this?

Reply:
I think eventually we would be timing this, but at the moment we're concerned about measuring the performance of the regridding performance because that has a bearing on using sparse vs scipy sparse .etc.

Reply:
wouldn't the prepare stage have a dependency on the choice of sparse implementation?  It should be fairly straight forward to add an additional `time_` method here to record the performance of the init method

Reply:
I suppose there would be _some_ effect on prepare, but I'd expect the effect to be minimal. The vast majority of the time spent in prepare will be spent in ESMF so that any changes to the performance of initialising the sparse matrix are going to be insignificant (unless something has gone very wrong).

Reply:
If we had a benchmark we could test your expectations, and have a baseline to tell us if something has gone very wrong... 😉

Reply:
at any rate, I think that half of the benchmarking is probably worth its own PR.

Reply:
ok fine, please create an issue to track adding this benchmark and add the issue to the backlog.